packages/template-retail-react-app/app/contexts/index.js

```diff
@@ -40,65 +40,53 @@ export const CategoriesProvider = ({treeRoot = {}, children, locale}) => {
 
     const api = useCommerceAPI()
     const [root, setRoot] = useState({
-        // map over the server-provided cat
+        // Clone the server-provided category
         ...treeRoot[DEFAULT_ROOT_CATEGORY],
         [itemsKey]: treeRoot[DEFAULT_ROOT_CATEGORY]?.[itemsKey]?.map((item) => ({
             ...item
         }))
     })
 
     const fetchCategoryNode = async (id, levels = 1) => {
+        const storageKey = `${LOCAL_STORAGE_PREFIX}${id}-${locale}`
+        const storageItem = JSON.parse(window.localStorage.getItem(storageKey))
+
         // return early if there's one that is less than stale time
-        const storageItem = JSON.parse(
-            window.localStorage.getItem(`${LOCAL_STORAGE_PREFIX}${id}-${locale}`)
-        )
-        if (storageItem || Date.now() < storageItem?.fetchTime + CAT_MENU_STALE_TIME) {
+        if (storageItem && Date.now() < storageItem.fetchTime + CAT_MENU_STALE_TIME) {
             return storageItem
         }
+        // get and store fresh data for faster access / reduced server load
+        // TODO: Convert to useCategory hook when hooks are ready
         const res = await api.shopperProducts.getCategory({
             parameters: {
                 id,
                 levels
             }
         })
+        res.loaded = true
+        window.localStorage.setItem(
```

Comment:
Moved this to inside the fetch function, so that fetch time is set on fetch, rather than on every page load.